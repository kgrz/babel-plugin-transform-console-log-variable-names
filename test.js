console.log(12);
console.log(12, 13, 14, 'asdfasdfdf');

const a = 'adsfsadfasd';
const b = 'asdfsadfsadfasf';
const c = 'asdfsadfsadfasf';
console.log(a);
console.log(a, b);
console.log(a, b, c);
