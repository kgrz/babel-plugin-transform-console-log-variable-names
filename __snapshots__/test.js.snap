// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds concatenated variable name as label to console.log argument 1`] = `
"
const a = 'wut';
const b = 'wut again';
const c = 'wut once again';
console.log('a, b, c: ', a, b, c);"
`;

exports[`adds correct label when printing out instance properties 1`] = `
"
this.a = 'wut';
console.log('this.a: ', this.a);"
`;

exports[`adds label when printing out functions 1`] = `
"
function a() {
		console.log('hi');
};
console.log('a: ', a);"
`;

exports[`adds variable name as label to console.log argument 1`] = `
"
const a = 'wut';
console.log('a: ', a);"
`;

exports[`does not add label if variables are not used 1`] = `
"
console.log('wut');"
`;

exports[`works inside class methods 1`] = `
"
class A {
	componentWillReceiveProps(nextProps) {
		console.log(\\"this.props: \\", this.props);
		console.log(\\"nextProps: \\", nextProps);
	}
}"
`;

exports[`works when printing variables pointing to functions 1`] = `
"
const wut = function () {
	console.log('hi');
};
console.log('wut: ', wut);"
`;

exports[`works when printing variables pointing to objects 1`] = `
"
const wut = { a: 1 };
console.log(\\"wut: \\", wut);"
`;
